Source/Csla.Blazor.WebAssembly.Tests/SessionManagerTests.cs

```diff
@@ -0,0 +1,205 @@
+
+
+           if (cancellationToken.IsCancellationRequested)
+           {
+             throw new OperationCanceledException(cancellationToken);
+           }
+           else
+           {
+             return new HttpResponseMessage()
+             {
+               StatusCode = HttpStatusCode.OK,
+               Content = new StringContent("{\"ResultStatus\":0, \"SessionData\":\"" + Convert.ToBase64String(GetSession(session, _applicationContext)) + "\"}"),
+             };
+           }
+         })
+         .Verifiable();
+
+      // use real http client with mocked handler here
+      var httpClient = new HttpClient(handlerMock.Object)
+      {
+        BaseAddress = new Uri("http://test.com/"),
+      };
+      return httpClient;
+    }
+
+    private static byte[] GetSession(SessionMessage session, ApplicationContext _applicationContext)
+    {
+      var info = new MobileFormatter(_applicationContext).SerializeToDTO(session);
+
+      var ms = new MemoryStream();
+
+      new CslaBinaryWriter(_applicationContext).Write(ms, info);
+
+      ms.Position = 0;
+      var array = ms.ToArray();
+      return array;
+    }
+
+    [TestMethod]
+    public async Task RetrieveSession_WithTimeoutValue_ShouldNotThrowException()
+    {
+      var timeout = TimeSpan.FromHours(1);
+      await _sessionManager.RetrieveSession(timeout);
+    }
+
+    [TestMethod]
+    public async Task RetrieveSession_WithZeroTimeout_ShouldNotThrowException()
+    {
+      var timeout = TimeSpan.Zero;
+      await Assert.ThrowsExceptionAsync<TaskCanceledException>(() => _sessionManager.RetrieveSession(timeout));
+
+    }
+
+    [TestMethod]
+    public async Task RetrieveSession_WithValidCancellationToken_ShouldNotThrowException()
+    {
+      var cts = new CancellationTokenSource();
+      await _sessionManager.RetrieveSession(cts.Token);
+    }
+
+    [TestMethod]
+    public async Task RetrieveSession_WithCancelledCancellationToken_ShouldThrowOperationCanceledException()
+    {
+      var cts = new CancellationTokenSource();
+      cts.Cancel();
+      await Assert.ThrowsExceptionAsync<TaskCanceledException>(() => _sessionManager.RetrieveSession(cts.Token));
+    }
+
+    [TestMethod]
+    public async Task SendSession_WithTimeoutValue_ShouldNotThrowException()
+    {
+      var timeout = TimeSpan.FromHours(1);
+      await _sessionManager.SendSession(timeout);
+    }
+
+    [TestMethod]
+    public async Task SendSession_WithZeroTimeout_ShouldNotThrowException()
+    {
+      var timeout = TimeSpan.Zero;
+      await Assert.ThrowsExceptionAsync<TaskCanceledException>(() => _sessionManager.SendSession(timeout));
+
+    }
+
+    [TestMethod]
+    public async Task SendSession_WithValidCancellationToken_ShouldNotThrowException()
+    {
+      var cts = new CancellationTokenSource();
+      await _sessionManager.SendSession(cts.Token);
+    }
+
+    [TestMethod]
+    public async Task SendSession_WithCancelledCancellationToken_ShouldThrowOperationCanceledException()
```

Comment:
Method name does not match test. If we were to change the exception thrown by `SendSession` from `TaskCanceledException` to `OperationCanceledException` this test would fail. `Assert.ThrowsExceptionsAsync` tests for the exact type of exception and you are testing for `TaskCanceledException` isntead of `OperationCanceledException` as the method name suggest.
Maybe you can take advantage of using [FluentAssertions](https://fluentassertions.com/exceptions/) here for easier checking?